pubsub/gossipsub/gossipsub-v1.1.md

```diff
@@ -0,0 +1,374 @@
+
+In order to implement PX, we extend the `PRUNE` control message to include an optional set of
+peers the pruned peer can connect to. This set of includes the Peer ID and a [_signed_ peer
+record](https://github.com/libp2p/specs/pull/217) for each peer exchanged.
+In order to facilitate transion to the usage of signed peer records within the libp2p ecosystem,
+the emitting peer is allowed to omit the signed peer record if it doesn't have one.
+In this case, the pruned peer will have to utilize an external service to discover addresses for
+the peer, eg the DHT.
+
+### Protobuf
+
+The `ControlPrune` message is extended with a `peer` field as follows.
+
+```protobuf
+message ControlPrune {
+	optional string topicID = 1;
+	repeated PeerInfo peers = 2; // gossipsub v1.1 PX
+}
+
+message PeerInfo {
+	optional bytes peerID = 1;
+	optional bytes signedPeerRecord = 2;
+}
+
+```
+
+## Flood Publishing
+
+In gossipsub v1.0 a freshly published message is propagated through the mesh or the fanout map
+if the publisher is not subscribed to the topic. In gossipsub v1.1 publishing is (optionally)
+done by publishing the message to all connected peers with a score above a publish threshold
+(see Peer Scoring below).
+
+This behaviour is prescribed to counter eclipse attacks and ensure that a newly published message
+from a honest node will reach all connected honest nodes and get out to the network at large.
+When flood publishing is in use there is no point in utilizing a fanout map or emitting gossip when
+the peer is a pure publisher not subscribed in the topic.
+
+This behaviour also reduces message propagation latency as the message is injected to more points
+in the network.
+
+## Adaptive Gossip Dissemination
+
+In gossipsub v1.0 gossip is emitted to a fixed number of peers, as specified by the `D_lazy`
+parameter. In gossipsub v1.1 the disemmination of gossip is adaptive; instead of emitting gossip
+to a fixed number of peers, we emit gossip to a percentage of our peers with a minimum of `D_lazy`
+peers.
+
+The parameter controlling the emission of gossip is called the gossip _factor_. When a node wants
+to emit gossip during the heartbeat, first it selects all peers with a peer score above a gossip
+threshold (see Peer Scoring below). From these peers, it randomly selects gossip factor peers with
+a minimum of `D_lazy`, and emits gossip to the selected peers.
+
+The recommended value for the gossip factor is `0.25`, which with the default of 3 rounds of gossip
+per message ensures that each peer has at least 50% chance of receiving gossip about a message.
+More specifically, for 3 rounds of gossip, the probability of a peer _not_ receiving gossip about
+a fresh message is `(3/4)³=27/64=0.421875`. So each peer receives gossip about a fresh message with
+a `0.578125` probability.
+
+This behaviour is prescribed to counter sybil attacks and ensures that a message from a honest
+node propagates in the network with high probability.
```

Comment:
This statement does not sound 100% accurate. Sybil attacks are not entirely "countered" with high probability. For instance, if 80% of the nodes are sybils and they invest enough time/work for the scoring function to reach its decay threshold, then they will be everywhere in the network, and then they can disconnect it. I think you mean something along the lines of: Sybil attacks whose resources are below a certain limit amount (of work, or something else) will be countered with high probability. And, of course, probably outside of the scope of a spec one would want to compute the exact amount of work that suffices to take down a network of a given size.

Reply:
Not sure where you are getting this 11% from. Even with 90% sybils, a little over than half the honest nodes will get the message via gossip (even if the mesh breaks down completely by being taken over by sybils somehow).
This half will propagate to the other half in a few hops at most, with high probability.